samples/tmo_shell/src/tmo_shell.c

```diff
@@ -128,6 +129,28 @@ int udp_cred_dtls(const struct shell *shell, size_t argc, char **argv);
 int udp_profile_dtls(const struct shell *shell, size_t argc, char **argv);
 #endif
 
+int read_hwid()
+{
+	struct adc_gecko_data *data = battery_dev->data;
+	int32_t buffer;
+	int err;
+	uint32_t millivolts = 0;
+
+	struct adc_sequence seq = {
+		.buffer = &buffer,
+		.buffer_size = sizeof(buffer),
+		.resolution = 12
+	};
+
+	seq.channels = 1;
+	err = adc_read(battery_dev, &seq);
+	if (err < 0) {
+		shell_print(shell,"Could not read (%d)\n", err);
+	}
+	millivolts = (uint32_t)(3.0 * ((data->mVolts * 2500.0) / 4096.0) + 0.5);
```

Comment:
This conversion shouldn't be applied to hwid

Reply:
Done